lib/worker.js

```diff
@@ -164,7 +164,7 @@ Worker.prototype._start = function() {
         self.serviceReturns = res;
         // Make sure that only JSON-serializable values are returned.
         try {
-            return [JSON.parse(JSON.stringify(res))];
+            return JSON.parse(JSON.stringify(res));
```

Comment:
But that's exactly what this whole patch fixes, you aren't doing that because you're pushing `msg.serviceReturns[0]`